lib/Driver/Compilation.cpp

```diff
@@ -513,8 +533,18 @@ namespace driver {
           // Parseable output was requested.
           FinishedCmd->forEachContainedJobAndPID(Pid, [&](const Job *J,
                                                           Job::PID P) {
-            parseable_output::emitFinishedMessage(llvm::errs(), *J, P,
-                                                  ReturnCode, Output, ProcInfo);
+            if (ReturnCode != 0 &&
+                isBatchJob(FinishedCmd) &&
+                batchConstituentJobProducedZeroLengthSerializedDiagnostics(J)) {
+              // Simulate SIGINT-interruption to parseable-output consumer for any
+              // constituent of a failing batch job that produced no errors of its own.
+              parseable_output::emitSignalledMessage(llvm::errs(), *J, P,
+                                                     "cancelled batch constituent",
```

Comment:
A string constant (`"cancelled batch constituent"`) probably belongs in a header file somewhere, in case it needs to be used somewhere else in the future.

Reply:
Hmm, nothing ought to sniff it and it's just a diagnostic hint for someone reading the parseable output; we have other diagnostic string constants sprinkled around. The only string constants I see in headers are builtin symbols / module names / filename extensions. I think this is ok as-is?

Reply:
Sure, fine to leave it. (Just a general feeling I have about string constants, about what may or may not happen in the future.)